cumulus/parachains/integration-tests/emulated/tests/bridges/bridge-hub-westend/src/tests/snowbridge_v2.rs

```diff
@@ -0,0 +1,1027 @@
+	});
+}
+
+#[test]
+fn send_token_to_penpal_v2() {
+	let relayer = BridgeHubWestendSender::get();
+	let relayer_location =
+		Location::new(0, AccountId32 { network: None, id: relayer.clone().into() });
+
+	let token: H160 = TOKEN_ID.into();
+	let token_location = erc20_token_location(token);
+
+	let beneficiary_acc_id: H256 = H256::random();
+	let beneficiary_acc_bytes: [u8; 32] = beneficiary_acc_id.into();
+	let beneficiary =
+		Location::new(0, AccountId32 { network: None, id: beneficiary_acc_id.into() });
+
+	let claimer_acc_id = H256::random();
+	let claimer = AccountId32 { network: None, id: claimer_acc_id.into() };
+	let claimer_bytes = claimer.encode();
+
+	// To pay fees on Penpal.
+	let eth_fee_penpal: xcm::prelude::Asset = (eth_location(), 3_000_000_000_000u128).into();
+
+	register_foreign_asset(eth_location());
+	register_foreign_asset(token_location.clone());
+
+	// To satisfy ED
+	PenpalB::fund_accounts(vec![(
+		sp_runtime::AccountId32::from(beneficiary_acc_bytes),
+		3_000_000_000_000,
+	)]);
+
+	let penpal_location = BridgeHubWestend::sibling_location_of(PenpalB::para_id());
+	let penpal_sovereign = BridgeHubWestend::sovereign_account_id_of(penpal_location);
+	PenpalB::execute_with(|| {
+		type RuntimeOrigin = <PenpalB as Chain>::RuntimeOrigin;
+
+		// Register token on Penpal
+		assert_ok!(<PenpalB as PenpalBPallet>::ForeignAssets::force_create(
+			RuntimeOrigin::root(),
+			token_location.clone().try_into().unwrap(),
+			penpal_sovereign.clone().into(),
+			true,
+			1000,
+		));
+
+		assert!(<PenpalB as PenpalBPallet>::ForeignAssets::asset_exists(
+			token_location.clone().try_into().unwrap(),
+		));
+
+		// Register eth on Penpal
+		assert_ok!(<PenpalB as PenpalBPallet>::ForeignAssets::force_create(
+			RuntimeOrigin::root(),
+			eth_location().try_into().unwrap(),
+			penpal_sovereign.clone().into(),
+			true,
+			1000,
+		));
+
+		assert!(<PenpalB as PenpalBPallet>::ForeignAssets::asset_exists(
+			eth_location().try_into().unwrap(),
+		));
+
+		assert_ok!(<PenpalB as Chain>::System::set_storage(
+			<PenpalB as Chain>::RuntimeOrigin::root(),
+			vec![(
+				PenpalCustomizableAssetFromSystemAssetHub::key().to_vec(),
+				Location::new(2, [GlobalConsensus(Ethereum { chain_id: CHAIN_ID })]).encode(),
+			)],
+		));
+	});
+
+	set_up_eth_and_dot_pool(eth_location());
+	set_up_eth_and_dot_pool_on_penpal(eth_location());
+
+	let token_transfer_value = 2_000_000_000_000u128;
+
+	let assets = vec![
+		// the token being transferred
+		NativeTokenERC20 { token_id: token.into(), value: token_transfer_value },
+	];
+
+	let token_asset: xcm::prelude::Asset = (token_location.clone(), token_transfer_value).into();
+	BridgeHubWestend::execute_with(|| {
+		type RuntimeEvent = <BridgeHubWestend as Chain>::RuntimeEvent;
+		let instructions = vec![
+			// Send message to Penpal
+			DepositReserveAsset {
+				// Send the token plus some eth for execution fees
+				assets: Definite(vec![eth_fee_penpal.clone(), token_asset].into()),
```

Comment:
this executes in AH context, coincidentally `eth` has same relative location from both AH and penpal, but for correctness this should be reanchored (in practice noop here) or at least named differently (even if same value)
```suggestion
				assets: Definite(vec![eth_penpal_fee_ah.clone(), token_asset_ah].into()),
```

Reply:
https://github.com/vgeddes/polkadot-sdk/blob/snowbridge-v2/cumulus/parachains/integration-tests/emulated/tests/bridges/bridge-hub-westend/src/tests/snowbridge_v2_inbound.rs#L540